File: 1.1/src/c/cz/mg/vulkan/PFN_vkDestroyBuffer.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyBuffer_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong buffer, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyBuffer f = (PFN_vkDestroyBuffer)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkBuffer*)jniLongToPointer(buffer)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkDestroyBuffer.java
package cz.mg.vulkan;

public class PFNvkDestroyBuffer extends VkFunctionPointer {
    public PFNvkDestroyBuffer() {
    }

    public PFNvkDestroyBuffer(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkDestroyBuffer(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyBuffer(long value) {
        setValue(value);
    }

    public PFNvkDestroyBuffer(VkInstance instance) {
        super(instance, new VkString("vkDestroyBuffer"));
    }

    public void call(VkDevice device, VkBuffer buffer, VkAllocationCallbacks pAllocator){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), buffer != null ? buffer.getVkAddress() : VkPointer.getNullAddress(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void call(long vkaddress, long device, long buffer, long pAllocator);
}
